python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py

```diff
@@ -0,0 +1,117 @@
+    the only supported type is ``ase.Atoms``."""
+
+    pass
+
+
+def systems_to_torch(
+    systems: Union[IntoSystem, List[IntoSystem]],
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    positions_requires_grad: bool = False,
+    cell_requires_grad: bool = False,
+) -> Union[System, List[System]]:
+    """Converts a system or a list of systems into a
+    ``metatensor.torch.atomistic.System`` or a list of such objects.
+
+    :param: systems: The system or list of systems to convert.
+    :param: dtype: The dtype of the output tensors. If ``None``, the default
+        dtype is used.
+    :param: device: The device of the output tensors. If ``None``, the default
+        device is used.
+    :param: positions_requires_grad: Whether the positions tensors of
+        the outputs should require gradients.
+    :param: cell_requires_grad: Whether the cell tensors of the outputs
+        should require gradients.
+
+    :return: The converted system or list of systems.
+    """
+
+    if isinstance(systems, list):
+        return [
+            _system_to_torch(
+                system, dtype, device, positions_requires_grad, cell_requires_grad
+            )
+            for system in systems
+        ]
+    else:
+        return _system_to_torch(
+            systems, dtype, device, positions_requires_grad, cell_requires_grad
+        )
+
+
+def _system_to_torch(
+    system: IntoSystem,
+    dtype: Optional[torch.dtype],
+    device: Optional[torch.device],
+    positions_requires_grad: bool,
+    cell_requires_grad: bool,
+) -> System:
+    """Converts a system into a ``metatensor.torch.atomistic.System``.
+
+    :param: system: The system to convert.
+    :param: dtype: The dtype of the output tensors. If ``None``, the default
+        dtype is used.
+    :param: device: The device of the output tensors. If ``None``, the default
+        device is used.
+    :param: positions_requires_grad: Whether the positions tensors of
+        the outputs should require gradients.
+    :param: cell_requires_grad: Whether the cell tensors of the outputs
+        should require gradients.
+
+    :return: The converted system.
+    """
+    if not HAS_ASE:
+        raise RuntimeError("The `ase` package is required to convert systems to torch.")
+
+    if not isinstance(system, ase.Atoms):
+        raise ValueError(
+            "Only `ase.Atoms` objects can be converted to `System`s. "
+            f"for now; got {type(system)}."
+        )
+
+    positions = torch.tensor(
+        system.positions,
+        requires_grad=positions_requires_grad,
+        dtype=dtype,
+        device=device,
+    )
+
+    if np.all(system.pbc):
+        cell = torch.tensor(
+            system.cell[:],
+            requires_grad=cell_requires_grad,
+            dtype=dtype,
+            device=device,
+        )
+    elif not np.any(system.pbc):
+        cell = np.zeros((3, 3))
```

Comment:
if the cell is not already zero, we send a warning in rascaline (to not modify the user data without their knowledge). Should we do the same here?

Reply:
Sure